job-runner/runner/runner.go

```diff
@@ -190,6 +188,34 @@ TRY_LOOP:
 	}
 }
 
+// Actually run the job.
+func (r *runner) runJob(jobData map[string]interface{}) (startTime, finishTime int64, ret job.Return, err error) {
+	var startedAt, finishedAt int64
+
+	defer func() {
+		// Recover from a panic inside Job.Run()
+		if panicErr := recover(); panicErr != nil {
+			// Set return values.
+			startTime = startedAt
```

Comment:
Why reassigning the value? Since it's set before running, it should always be set.